taichi/codegen/codegen_llvm.cpp

```diff
@@ -1605,13 +1613,24 @@ void CodeGenLLVM::create_offload_struct_for(OffloadedStmt *stmt, bool spmd) {
     builder->SetInsertPoint(loop_body_bb);
 
     // initialize the coordinates
-    auto refine =
-        get_runtime_function(leaf_block->refine_coordinates_func_name());
     auto new_coordinates = create_entry_block_alloca(physical_coordinate_ty);
 
     create_call(refine, {parent_coordinates, new_coordinates,
                          builder->CreateLoad(loop_index)});
 
+    if (stmt->snode->type == SNodeType::bit_array && stmt->snode->parent) {
```

Comment:
Could you explain why is this needed?

Reply:
One more refine step was needed for bit_arrays to make the loop coordinates non-consecutive since the operation is going to be vectorized over multiple coordinates per loop_index. 
For example, a 1-d block of 64 leaf elements and vectorization of 32, only two loop coordinates are needed. Without this extra refine step, their coordinates would be a consecutive 0, 1 instead of non-consecutive 0, 32. 
This step was also not needed before since coordinates were always non-consecutive.